server/common/rest.py

```diff
@@ -215,6 +306,135 @@ def inflate(data):
     return zlib.decompress(data)
 
 
+def annotations_obs_put(request, data_adaptor):
```

Comment:
Write path: requires resolved user_id. Please verify error codes and failure modes so autosave can surface issues cleanly; consider retry/backoff policy.